### ISSUE TITLE
Audit all runtime tests that use regex for output parsing

### ISSUE THREAD

**Issue by danobi** [bpftrace/bpftrace#2993]
We've noticed quite a few runtime tests that use regex to validate output are overly permissive in accepted input or are flat out wrong. For example `other.positional lhist` is probably wrong.

Would be good to audit all the runtime tests that use regex to check if they can be replaced with something a little easier to get right.

See https://github.com/bpftrace/bpftrace/pull/2989 for some more details.

**Comment by ajor** [comment]
Also tests with just: `EXPECT Attaching 1 probe...`. These only check that the program can run and not that its output is correct.

**Comment by shimarda** [comment]
Hi, I’d like to work on this issue.

I’ll start by auditing the obviously broad `EXPECT_REGEX` usages in `tests/runtime`, especially cases like `EXPECT_REGEX .*` that do not seem to validate the intended output.

My initial plan is to keep the scope small and submit a PR for a subset of runtime tests first, likely starting with `tests/runtime/other`. For cases where regex is still needed due to environment-dependent output, I’ll try to tighten the pattern rather than replacing it with `EXPECT`.

Please let me know if there is a preferred subset or approach I should start with.

**Comment by jordalgo** [comment]
@shimarda Thanks for volunteering! You are free to work on this but if you want something that is higher-priority (in this same space) it would be migrating runtime tests to self tests:
https://github.com/bpftrace/bpftrace/tree/master/tests#self-tests

Not all runtime tests (like those that test printing) can be migrated to self tests but there are probably many others that can be migrated which will really help CI run times. 

Does that sound interesting?

**Comment by shimarda** [comment]
Thanks for the suggestion!

Yes, that sounds interesting. I’d be happy to look into migrating suitable runtime tests to self tests as a higher-priority task.

I’ve also already found a small cleanup for this issue: a few `EXPECT_REGEX .*` cases in `tests/runtime/other` can be tightened to `EXPECT_REGEX_NONE @`. Would it be okay to submit that as a small PR in parallel?

For the migration work, I’ll start by reading the self test docs and looking for runtime tests that do not depend on output formatting or CLI behavior. If there is a good starting point you recommend, please let me know.

**Comment by jordalgo** [comment]
> Would it be okay to submit that as a small PR in parallel?

Of course! Please do.

> If there is a good starting point you recommend, please let me know.

Not sure offhand but just take a look at some of the runtime test suites. You can see that I opted for self tests when I added the new record types:
https://github.com/bpftrace/bpftrace/blob/master/tests/self/record.bt

**Comment by shimarda** [comment]
Thanks, I’ll submit the small PR for the `EXPECT_REGEX .*` cleanup first.

I’ll also take a look at `tests/self/record.bt` and then go through some runtime test suites to find small cases that can be migrated to self tests. I’ll avoid tests that depend on output formatting or CLI behavior.
For the runtime-to-self test migration, would you prefer that I open a separate issue/PR discussion once I identify a small candidate test suite?
